### DISCUSSION TITLE
Hide radix imports

### DISCUSSION THREAD

**Discussion by dominikjasek** [shadcn-ui/ui#2535] [General]
Hey, is there any eslint or tsconfig rule which forbids default radix components?

In the following picture, I don't want to see second row.

<img width="759" alt="image" src="https://github.com/shadcn-ui/ui/assets/48070343/320f5c95-84d3-48d6-915f-bf0a56aeb8a3">

**Comment by Alexjoshua14** [comment]
I'm not sure if this is directly helpful for you but for anyone using VSCode, you can add '@radix-ui' to the `Auto Import File Exclude Patterns` option in VSCode's settings.

**Reply by dominikjasek** [reply]
Thanks, for Intellij IDEs you can do this:
![image](https://github.com/shadcn-ui/ui/assets/48070343/579b1267-8034-44cb-9ad2-12c1d7dce9ef)

**Reply by NjengaFelix** [reply]
It hasn't worked on my side this is how I have configured vscode 

### On file->preferences -> settings
![image](https://github.com/shadcn-ui/ui/assets/32095327/eaf1880e-ad67-4dcf-bb08-e95dc52ca5b4)

![image](https://github.com/shadcn-ui/ui/assets/32095327/75c06e6c-b6f8-4fa4-9e0f-b2e135a6d43e)

**Reply by Alexjoshua14** [reply]
@NjengaFelix  I could be wrong but I think that the problem there is the quotes that you have surrounding @radix-ui. I think that currently your setting is set to exclude `'@radix-ui'` instead of `@radix-ui`. Here's a screen shot of the exact setting I have that works for my VSCode environment. If this doesn't resolve your issue, let me know and I'll see if there's anything else I can do to help.
<img width="978" alt="Screenshot 2024-01-25 at 10 59 02 AM" src="https://github.com/shadcn-ui/ui/assets/59298565/6a4e78a7-33da-4288-9701-24a09088cfd5">

**Reply by rwieruch** [reply]
JSON Settings

```
  "typescript.preferences.autoImportFileExcludePatterns": [
    "@radix-ui"
  ],
```

https://twitter.com/rwieruch/status/1767628375190102301

**Reply by NjengaFelix** [reply]
![image](https://github.com/shadcn-ui/ui/assets/32095327/922756b2-3159-4b8a-8c3c-73b620b54ea6)

Thank you this works

**Reply by joaopedrodcf** [reply]
@rwieruch that's actually pretty nice!

**Reply by zenzen-sol** [reply]
This also blocks auto-import of `@radix-ui/react-icons`, and I guess anything else in the `@radix-ui` namespace that doesn't have a counterpart in shadcn-ui.

@rwieruch I replied to your tweet with an incorrect clarification. I assumed that normal glob rules would apply in the exclude patterns, but they don't seem to work.

Does anyone have a clever trick for allowing only `@radix-ui/react-icons`, but blocking all the rest of `@radix-ui/*`?

**Reply by mizrabsheikh** [reply]
@zenzen-sol Did you find any workaround for allowing specific imports?

**Reply by zenzen-sol** [reply]
@mizrabsheikh No, I was never able to find a solution for this, unfortunately. No matter what I did with the exclude pattern, there is either a blindspot for react-icons, or you catch all of radix-ui.

**Reply by chrisribal** [reply]
> This also blocks auto-import of `@radix-ui/react-icons`, and I guess anything else in the `@radix-ui` namespace that doesn't have a counterpart in shadcn-ui.
> 
> @rwieruch I replied to your tweet with an incorrect clarification. I assumed that normal glob rules would apply in the exclude patterns, but they don't seem to work.
> 
> Does anyone have a clever trick for allowing only `@radix-ui/react-icons`, but blocking all the rest of `@radix-ui/*`?

Try to set the exclude rule to `@radix-ui/!(icons)`. This excludes every radix-ui package, but the icons.

**Reply by vhbui02** [reply]
> > This also blocks auto-import of `@radix-ui/react-icons`, and I guess anything else in the `@radix-ui` namespace that doesn't have a counterpart in shadcn-ui.
> > @rwieruch I replied to your tweet with an incorrect clarification. I assumed that normal glob rules would apply in the exclude patterns, but they don't seem to work.
> > Does anyone have a clever trick for allowing only `@radix-ui/react-icons`, but blocking all the rest of `@radix-ui/*`?
> 
> Try to set the exclude rule to `@radix-ui/!(icons)`. This excludes every radix-ui package, but the icons.

You're doing it wrong. It's `"typescript.preferences.autoImportFileExcludePatterns": [
    "@radix-ui/react-a*","@radix-ui/react-o*", 
    "@radix-ui/react-b*","@radix-ui/react-p*",
    "@radix-ui/react-d*","@radix-ui/react-q*",
    "@radix-ui/react-c*","@radix-ui/react-r*",
    "@radix-ui/react-e*","@radix-ui/react-s*",
    "@radix-ui/react-f*","@radix-ui/react-t*",
    "@radix-ui/react-g*","@radix-ui/react-u*",
    "@radix-ui/react-h*","@radix-ui/react-v*",
    "@radix-ui/react-j*","@radix-ui/react-w*",
    "@radix-ui/react-k*","@radix-ui/react-x*",
    "@radix-ui/react-l*","@radix-ui/react-y*",
    "@radix-ui/react-m*","@radix-ui/react-z*",
    "@radix-ui/react-n*",
  ],`

**Reply by longzheng** [reply]
I used ChatGPT to generate a list of all the Radix imports that Shadcn components use
```ts
{
    "typescript.preferences.autoImportFileExcludePatterns": [
        "/node_modules/storybook/**/*",
        "/src/**/*.stories.tsx",
        "@radix-ui/react-accordion",
        "@radix-ui/react-alert-dialog",
        "@radix-ui/react-aspect-ratio",
        "@radix-ui/react-avatar",
        "@radix-ui/react-checkbox",
        "@radix-ui/react-collapsible",
        "@radix-ui/react-context-menu",
        "@radix-ui/react-dialog",
        "@radix-ui/react-dropdown-menu",
        "@radix-ui/react-hover-card",
        "@radix-ui/react-label",
        "@radix-ui/react-menubar",
        "@radix-ui/react-navigation-menu",
        "@radix-ui/react-popover",
        "@radix-ui/react-progress",
        "@radix-ui/react-radio-group",
        "@radix-ui/react-scroll-area",
        "@radix-ui/react-select",
        "@radix-ui/react-separator",
        "@radix-ui/react-slider",
        "@radix-ui/react-switch",
        "@radix-ui/react-tabs",
        "@radix-ui/react-toast",
        "@radix-ui/react-toggle",
        "@radix-ui/react-toggle-group",
        "@radix-ui/react-tooltip"
    ]
}
```

**Reply by vhbui02** [reply]
> I used ChatGPT to generate a list of all the Radix imports that Shadcn components use
> 
> ```ts
> {
>   "typescript.preferences.autoImportFileExcludePatterns": [
>     "@radix-ui/react-accordion",
>     "@radix-ui/react-alert-dialog",
>     "@radix-ui/react-aspect-ratio",
>     "@radix-ui/react-avatar",
>     "@radix-ui/react-checkbox",
>     "@radix-ui/react-collapsible",
>     "@radix-ui/react-context-menu",
>     "@radix-ui/react-dialog",
>     "@radix-ui/react-dropdown-menu",
>     "@radix-ui/react-hover-card",
>     "@radix-ui/react-menubar",
>     "@radix-ui/react-navigation-menu",
>     "@radix-ui/react-popover",
>     "@radix-ui/react-progress",
>     "@radix-ui/react-radio-group",
>     "@radix-ui/react-scroll-area",
>     "@radix-ui/react-select",
>     "@radix-ui/react-separator",
>     "@radix-ui/react-slider",
>     "@radix-ui/react-switch",
>     "@radix-ui/react-tabs",
>     "@radix-ui/react-toggle-group",
>     "@radix-ui/react-toast",
>     "@radix-ui/react-tooltip"
>   ]
> }
> ```

Please don't use ChatGPT to handle these task:
- First, it's a "near-100%" accuracy pattern matcher, but it will never be 100%. So there can be mistakes.
- Second, the model's training date cut-off is dated back to a few months. It might not reflect newer changes on shadCN's upstream version.
